Release v2.15.0 to master

laurent-laporte-pro opened the pull request:
Next minor release v2.15.0 on `master`

MartinBelthle committed:
fix(command): fix update_config to support API PUT v1/studies/{uuid}/raw

MartinBelthle committed:
fix(command): change code according to comments

MartinBelthle committed:
feat(storage): ini reader can now take ioTextWrapper as input

laurent-laporte-pro committed:
feat(matrix): improve matrix read/write using NumPy

laurent-laporte-pro committed:
fix(variant-manager): correct the variant manager

laurent-laporte-pro committed:
refactor(allocation): remove deprecated class `FixedAllocationKeyIniReader`

laurent-laporte-pro committed:
test(ini): correct the `IniFileNode` unit tests

laurent-laporte-pro committed:
fix(api): correct the regex in `IniReader` to match a section in square brackets

laurent-laporte-pro committed:
style: reindent source code in unit tests

laurent-laporte-pro committed:
Merge pull request #1505 from AntaresSimulatorTeam/release/v2.14

v2.14 on dev

laurent-laporte-pro committed:
refactor(rawstudy-filesystem): improve error message raised by `ChildNotFoundError`

laurent-laporte-pro committed:
test(matrix-service): improve unit tests for `MatrixService` class

laurent-laporte-pro committed:
feat(matrix): improve matrix service read/write using NumPy

laurent-laporte-pro committed:
Merge pull request #1516 from AntaresSimulatorTeam/hotfix/2.14.1

hotfix/2.14.1 on dev

laurent-laporte-pro committed:
style: reindent `ini_reader.py`

laurent-laporte-pro committed:
test(matrix-service): improve unit tests for `MatrixContentRepository` class

hdinia committed:
feat(ui-map): add map zoom buttons (#1518)

laurent-laporte-pro committed:
test(matrix): use NumPy arrays in `parse_commands` UT and correct path on Windows

MartinBelthle committed:
Merge pull request #1461 from AntaresSimulatorTeam/fix/1289-bis-api-updtae-config

MartinBelthle committed:
fix(web): change API response_model to avoid ValidationError (#1526)

skamril committed:
feat(common): multiple updates on Form component

* support loading for manual submit
* update effects
* add `setValues` in form API
* create useFormApiPlus

MartinBelthle committed:
feat(model): handle 8.6 study model and study upgrader (#1489)

Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>

laurent-laporte-pro committed:
refactor(matrix-service): use `matrix_id` in favor of `matrix_hash` as matrix hashing is an implementation detail

laurent-laporte-pro committed:
refactor(matrix-service): simplify timestamp calculation used during `MatrixDTO` creation

laurent-laporte-pro committed:
feat(matrix-service): enhance the `MatrixContentRepository` class to support saving NumPy array directly

laurent-laporte-pro committed:
style(matrix-service): correct the Google docstring format

laurent-laporte-pro committed:
feat(matrix-service): enhance the `ISimpleMatrixService` class and sub-classes to support creating from NumPy array

laurent-laporte-pro committed:
feat(matrix-service):  improve the importation function to handle NumPy arrays directly without the necessity of converting them

laurent-laporte-pro committed:
feat(matrix-service): avoid having to save the matrix again

laurent-laporte-pro committed:
fix(matrix-service): correct the `MatrixService.create` so that the matrix file is not deleted in case of exception

skamril committed:
feat(ui): enable undo/redo on some forms

skamril committed:
feat(common): add undo/redo to Form component

laurent-laporte-pro committed:
feat(matrix-service): ensure exclusive access to the matrix file between multiple processes (or threads)

laurent-laporte-pro committed:
fix(matrix-service): prevent the use of scientific notation when saving matrices

MartinBelthle committed:
feat(business): add xpansion batch size back office (#1485)

Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>

laurent-laporte-pro committed:
fix: correct the import of `declarative_base` to avoid a deprecation warning.

laurent-laporte-pro committed:
refactor: enhance code quality in variant commands implementation

skamril committed:
fix(ui-i18n): add missing key

skamril committed:
feat(api): add new pollutants in thermal manager

laurent-laporte-pro committed:
test(api): simplify initialization of the database in unit tests (#1540)

Co-authored-by: BELTHLE Martin <martin.belthle@rte-france.com>

skamril committed:
feat(ui-thermal): add new pollutants

skamril committed:
Merge pull request #1497 from AntaresSimulatorTeam/feature/1480-add-pollutants-to-thermal-form

skamril committed:
Merge pull request #1536 from AntaresSimulatorTeam/feature/1492-add-undoredo-feature-on-forms

skamril committed:
feat(ui-config): enable undo/redo in adequacy patch form (#1543)

laurent-laporte-pro committed:
test: improve and correct unit tests for `AreaManager`

laurent-laporte-pro committed:
fix(storage): fix INI file writing anomaly in Antares study update (#1542)

laurent-laporte-pro committed:
refactor(api-model): improve `transform_name_to_id` implementation and add UT

laurent-laporte-pro committed:
Merge pull request #1452 from AntaresSimulatorTeam/feature/1451-improve-TS-read-write-performance

feat(matrix): improve matrix read/write using NumPy

laurent-laporte-pro committed:
Merge pull request #1539 from AntaresSimulatorTeam/feat/1525-enhance_variant_commands_impl

refactor(variant): enhance implementation of variant commands

skamril committed:
build: specify the node versions supported  (#1553)

hdinia committed:
fix(api): standardize 'areas_ui' to dict format for single area case (#1557)

Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>

laurent-laporte-pro committed:
Merge pull request #1546 from AntaresSimulatorTeam/feature/refactoring-transform_name_to_id

refactor(api-model): improve `transform_name_to_id` implementation for more efficiency

sylvlecl committed:
fix(build): fix pyinstaller build  (#1551)

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>
Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>

sylvlecl committed:
fix(worker): archive worker must be kept alive for processing (#1558)

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>
Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>
Co-authored-by: Laurent LAPORTE <43534797+laurent-laporte-pro@users.noreply.github.com>

laurent-laporte-pro committed:
style(ui): correct spelling mistake in webapp code (#1570)

skamril committed:
fix(api,tablemode): wrong adequacy patch mode path and clean code

skamril committed:
feat(ui-common): update ColorPickerFE

skamril committed:
fix(ui-model): update Properties form with the new endpoint

skamril committed:
fix(ui-i18n): missing key

skamril committed:
style(api): change field version check

skamril committed:
test: add Properties form test and fix others one

skamril committed:
feat(api): add Properties form endpoint

skamril committed:
Merge pull request #1580 from AntaresSimulatorTeam/bugfix/1537_properties_form

skamril committed:
Merge branch 'hotfix/v2.14.2' into dev

skamril committed:
Merge branch 'hotfix/v2.14.3' into dev

laurent-laporte-pro committed:
fix(filesystem): correct the handling of default matrices in `InputSeriesMatrix` node (#1608)

laurent-laporte-pro committed:
build(launcher): use Antares-Launcher v1.3.0 which preserve the "output" directory required by the `--step=sensitivity` Xpansion option (#1606)

MartinBelthle committed:
fix(st-storage): replace inflow.txt by inflows.txt according to the changing doc (#1618)

Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>

laurent-laporte-pro committed:
fix(api): study data retrieval now considers document types, enabling retrieval of Microsoft Office documents. Binary data support has also been enhanced

MartinBelthle committed:
fix(service): set public mode to `NONE` and assign user groups to the study during copying (#1620)

MartinBelthle committed:
fix(service): set public mode to `NONE` and assign user groups to the study during import (#1619)

laurent-laporte-pro committed:
Merge branch 'hotfix/v2.14.4' into dev

# Conflicts:
#	antarest/matrixstore/model.py
#	antarest/study/business/xpansion_management.py

MartinBelthle committed:
feat(service): set the user's name as author in `study.antares` when creating a study (#1632)

Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>

laurent-laporte-pro committed:
refactor(command): improve implementation of area management (#1636)

laurent-laporte-pro committed:
feat(matrix): improve unexpected exception error message (#1635)

laurent-laporte-pro committed:
test: improve fixtures in unit tests for better reuse (#1638)

laurent-laporte-pro committed:
docs: correct and improve the "Variant Manager" documentation (#1637)

laurent-laporte-pro committed:
refactor(command): rename "cluster" as "thermal cluster" (#1639)

MartinBelthle committed:
feat(service): set PublicMode to READ when importing is done by user with no groups

laurent-laporte-pro committed:
refactor: refactoring of unit tests and Model improvement (#1647)

MartinBelthle committed:
feat(api): add api doc and integration test

laurent-laporte-pro committed:
fix(packaging): update the packaging script to use Antares Solver v8.6.1 (#1652)

laurent-laporte-pro committed:
feat(commands): add ST-Storage commands (#1630)

MartinBelthle committed:
fix(export): ZIP outputs are now uncompressed before export (#1656)

hdinia committed:
fix(api): add missing use_leeway field and validation rules (#1650)

skamril committed:
fix(ui-common): error with formState in Form

skamril committed:
feat(ui-common): add submitButtonIcon prop in Form

skamril committed:
feat(ui-login): update form

skamril committed:
feat(ui-common): display submit error in Form

skamril committed:
Merge pull request #1660 from AntaresSimulatorTeam/bugfix/login_submit

MartinBelthle committed:
fix(st_storage): add default value for st storage group

MartinBelthle committed:
fix(tests): use deepcopy for dict to only modify the copy

MartinBelthle committed:
feat(api): improves "launcher/_versions" endppoint

MartinBelthle committed:
fix(st_storage): parse_st_storage works with zipped study with version previous 860

hdinia committed:
fix(ui-map): area persistence issue after deletion

hdinia committed:
feat(ui-common): increase the width of the first column on tables

hdinia committed:
fix(api-layers): fixed issue preventing layers deletion when no areas

hdinia committed:
fix(ui-redux): missing state update for layers/districts after node deletion

hdinia committed:
feat(ui-i18n): add missing adequacy patch translations (#1680)

hdinia committed:
tests(api-layers): add layers management tests

hdinia committed:
style(ui-redux-map): minor improvements

hdinia committed:
feat(api-layers): minor improvements

hdinia committed:
fix(ui-redux): remove eslint no-param-reassign warning

hdinia committed:
fix(api-layers): remove unnecessary parameter that causes tests to fail

laurent-laporte-pro committed:
ci: upgrade mypy to v1.4.1 and Black to v23.7.0 for improved typing and formatting (#1685)

hdinia committed:
Merge pull request #1677 from AntaresSimulatorTeam/bugfix/1565-defects-in-area-layers-districts

fix(ui-api): resolve Area deletion & Layers/Districts UI modification issues

MartinBelthle committed:
fix(log-parser): simplify parsing and make it accurate (#1682)

Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>

laurent-laporte-pro committed:
fix(variant): enhance `VariantCommandsExtractor` for Renewable Clusters and Short-Term Storage (#1688)

MartinBelthle committed:
fix(xpansion): correct field names and types in Xpansion settings (#1684)

Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>

laurent-laporte-pro committed:
fix(api): correct response model of "/studies/_versions" endpoint

laurent-laporte-pro committed:
fix(api): correct exception handling in `extract_file_to_tmp_dir`

laurent-laporte-pro committed:
refactor(api): refactor data extraction logic in `_extract_data_from_file` function

- Add a concise docstring to the `_extract_data_from_file` function.
- Implement case-insensitive file suffix comparison to ensure robust file type detection.
- Consistently use UTF-8 encoding instead of platform-dependent encoding.
- Replace the custom `FileTypeNotSupportedException` exception with the standard `NotImplementedError`, simplifying the code.
- Refactor the implementation for shorter and more maintainable code, enhancing readability.

laurent-laporte-pro committed:
style: correct mypy issues

laurent-laporte-pro committed:
Merge pull request #1673 from AntaresSimulatorTeam/fix/we_cannot_longer_unarchive_study

fix(st-storage): fixing archived study configuration reading for short-term storages

laurent-laporte-pro committed:
feat(api): change the response model of "launcher/_versions" endpoint to return a list of versions

laurent-laporte-pro committed:
feat(api): improve exception handling in "launcher/_versions" endpoint

hdinia committed:
feat(ui): add and use correct launcher verisons instead of study's

laurent-laporte-pro committed:
feat(api): improve error handling in "launcher/_versions" endpoint

laurent-laporte-pro committed:
feat(api): replace "launcher/_versions" endpoint by "/launcher/versions"

laurent-laporte-pro committed:
feat(api): correct default values for "/launcher/versions" endpoint

laurent-laporte-pro committed:
fix(st-storage): correct docstring

laurent-laporte-pro committed:
test(st-storage): improve unit tests to check "group" default value and add typing

laurent-laporte-pro committed:
Merge pull request #1671 from AntaresSimulatorTeam/improves_endpoint_solver_versions

fix(launcher): fixing launcher versions display and API endpoint for solver versions

hdinia committed:
Merge pull request #1665 from AntaresSimulatorTeam/fix/add_default_value_for_st_storage_group

fix(st-storage): update "group" parameter values for Short-Term Storage

skamril committed:
fix(ui-common): prevent null values for string cells in HandsonTable component (#1689)

MartinBelthle committed:
fix(table-mode): fixes the reading of UI information in the case where the study has only one area  (#1674)

Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>

laurent-laporte-pro committed:
fix(table-mode): issue to read area information in the case where the study has only one area (#1690)

laurent-laporte-pro committed:
fix(variant): fixed TS deletion of renewable clusters and short-term storage (#1693)

hdinia committed:
feat(ui-study): enhance study upgrade dialog (#1687)

Co-authored-by: Samir Kamal <1954121+skamril@users.noreply.github.com>

hdinia committed:
feat(ui-lib): add material-react-table package

hdinia committed:
feat(ui-common): add GroupedDataTable

hdinia committed:
chore(github-actions): update Node.js version to 18.16.1

hdinia committed:
Merge pull request #1633 from AntaresSimulatorTeam/feature/1513-new-design-for-listings-clusters

feat(ui-common): add GroupedDataTable

laurent-laporte-pro committed:
Merge branch 'hotfix/v2.14.5' into dev

# Conflicts:
#	antarest/launcher/web.py
#	antarest/study/storage/rawstudy/model/filesystem/ini_file_node.py
#	antarest/study/storage/rawstudy/model/filesystem/matrix/input_series_matrix.py
#	antarest/study/storage/variantstudy/model/command/create_area.py
#	antarest/study/storage/variantstudy/model/command/create_renewables_cluster.py
#	antarest/study/storage/variantstudy/model/command/icommand.py
#	antarest/study/storage/variantstudy/model/command/remove_renewables_cluster.py
#	antarest/study/storage/variantstudy/variant_command_extractor.py
#	tests/integration/assets/launcher_mock.sh
#	tests/integration/conftest.py
#	tests/integration/test_studies_upgrade.py
#	tests/integration/variant_blueprint/test_renewable_cluster.py
#	tests/launcher/test_service.py
#	tests/storage/integration/test_exporter.py
#	webapp/package-lock.json

laurent-laporte-pro committed:
test(export): correct unit tests for `VariantStudyService.generate_task` method

TLAIDI committed:
feat(st-storage): add ST Storage API endpoints (#1697)

feat(st-storage): add ST Storage API endpoints (#1697)

MartinBelthle committed:
fix(model): wrong frequency for `hydro energy credits` matrices (#1708)

laurent-laporte-pro committed:
style: add configuration for iSort

laurent-laporte-pro committed:
style: ignore unused imports in `dbmodel.py`

laurent-laporte-pro committed:
chore: increase line length limit to 120 characters for Black and iSort

laurent-laporte-pro committed:
build: add iSort in the `requirements-dev.txt`

laurent-laporte-pro committed:
ci: change the main GitHub action to run iSort for code style checking

laurent-laporte-pro committed:
ci: correct the main GitHub action to use `Black~=23.7.0`

laurent-laporte-pro committed:
refactor: correct SonarCloud issues

laurent-laporte-pro committed:
refactor: fix lambda capturing `name` variable reported by SonarCloud

laurent-laporte-pro committed:
refactor: avoid always returning the same value (reported by SonarCloud)

laurent-laporte-pro committed:
style: sort imports in project base code and unit tests using iSort

laurent-laporte-pro committed:
style: remove unused imports in production base code

laurent-laporte-pro committed:
style: remove unused imports in unit tests

laurent-laporte-pro committed:
style: sort imports in project base code and unit tests using `line_length = 120`

laurent-laporte-pro committed:
chore: remove `# fmt: off`/`# fmt: on` Black directives

laurent-laporte-pro committed:
style: reindent project base code, `scripts/` and unit tests using `line-length = 120`

TLAIDI committed:
fix(st-storage): allow DELETE endpoint to accept multiple IDs as parameters (#1716)

fix(st-storage): allow delete endpoint to accept multiple IDs as parameters #1716

laurent-laporte-pro committed:
feat(style): code style and refactoring enhancements

laurent-laporte-pro committed:
Merge remote-tracking branch 'origin/hotfix/v2.14.6' into dev

# Conflicts:
#	.github/workflows/main.yml
#	antarest/core/tasks/model.py
#	antarest/matrixstore/matrix_garbage_collector.py
#	antarest/study/business/area_management.py
#	antarest/study/business/utils.py
#	antarest/study/business/xpansion_management.py
#	antarest/study/service.py
#	antarest/study/storage/abstract_storage_service.py
#	antarest/study/storage/rawstudy/model/filesystem/config/files.py
#	antarest/study/storage/rawstudy/model/filesystem/matrix/input_series_matrix.py
#	antarest/study/storage/rawstudy/model/filesystem/root/input/hydro/series/area/area.py
#	antarest/study/storage/rawstudy/model/filesystem/root/input/input.py
#	antarest/study/storage/rawstudy/model/filesystem/root/input/load/load.py
#	antarest/study/storage/rawstudy/model/filesystem/root/input/solar/solar.py
#	antarest/study/storage/rawstudy/model/filesystem/root/settings/generaldata.py
#	antarest/study/storage/rawstudy/raw_study_service.py
#	antarest/study/storage/storage_service.py
#	antarest/study/storage/study_upgrader/__init__.py
#	antarest/study/storage/variantstudy/business/command_extractor.py
#	antarest/study/storage/variantstudy/business/command_reverter.py
#	antarest/study/storage/variantstudy/business/matrix_constants/__init__.py
#	antarest/study/storage/variantstudy/business/utils.py
#	antarest/study/storage/variantstudy/command_factory.py
#	antarest/study/storage/variantstudy/model/command/create_area.py
#	antarest/study/storage/variantstudy/model/command/create_cluster.py
#	antarest/study/storage/variantstudy/model/command/icommand.py
#	antarest/study/storage/variantstudy/model/command/remove_area.py
#	antarest/study/storage/variantstudy/model/command/remove_cluster.py
#	antarest/study/storage/variantstudy/variant_command_extractor.py
#	antarest/study/storage/variantstudy/variant_study_service.py
#	antarest/study/web/raw_studies_blueprint.py
#	antarest/study/web/study_data_blueprint.py
#	antarest/study/web/variant_blueprint.py
#	antarest/tools/lib.py
#	tests/conftest.py
#	tests/integration/utils.py
#	tests/launcher/test_extension_adequacy_patch.py
#	tests/launcher/test_log_parser.py
#	tests/matrixstore/conftest.py
#	tests/storage/business/test_arealink_manager.py
#	tests/storage/business/test_raw_study_service.py
#	tests/storage/business/test_study_version_upgrader.py
#	tests/storage/conftest.py
#	tests/storage/integration/conftest.py
#	tests/storage/integration/test_STA_mini.py
#	tests/storage/repository/filesystem/config/test_config_files.py
#	tests/storage/repository/filesystem/config/test_utils.py
#	tests/storage/repository/filesystem/root/input/hydro/common/capacity/test_capacity.py
#	tests/storage/repository/filesystem/root/output/simulation/mode/common/test_area.py
#	tests/storage/repository/test_study.py
#	tests/study/business/test_allocation_manager.py
#	tests/study/business/test_correlation_manager.py
#	tests/variantstudy/conftest.py
#	tests/variantstudy/model/command/test_manage_binding_constraints.py
#	tests/variantstudy/model/command/test_replace_matrix.py
#	tests/variantstudy/model/test_variant_model.py
#	tests/variantstudy/test_command_factory.py
#	webapp/package-lock.json

laurent-laporte-pro committed:
build: prepare next minor release v2.15.0 (unreleased)

ghost commented:
### Features

*  **ui-map:** add map zoom buttons (#1518) ([04cd3a5](https://github.com/AntaresSimulatorTeam/AntaREST/commit/04cd3a5cb6df8730035abf075829165305bd6dae))
*  **storage:** ini reader can now take ioTextWrapper as input ([015bc12](https://github.com/AntaresSimulatorTeam/AntaREST/commit/015bc12a5e1530144133cb072633a8ec4f2e3e23))
*  **model:** handle 8.6 study model and study upgrader (#1489) ([9d091e8](https://github.com/AntaresSimulatorTeam/AntaREST/commit/9d091e85b6a9d5aa562fdaba1b6f7bd4d6aa3db8))
*  **business:** add xpansion batch size back office (#1485) ([a0b2cfb](https://github.com/AntaresSimulatorTeam/AntaREST/commit/a0b2cfbab8324e96fa47a49bd8d80f2617adf868))
*  **api:** add new pollutants in thermal manager ([bcd5a01](https://github.com/AntaresSimulatorTeam/AntaREST/commit/bcd5a01e6878048512550d26834d4dad335b975e))
*  **ui-thermal:** add new pollutants ([5164a4c](https://github.com/AntaresSimulatorTeam/AntaREST/commit/5164a4cb5a894f5b96f6c0fd21b358e4df072aec))
*  **common:** multiple updates on Form component ([2193e43](https://github.com/AntaresSimulatorTeam/AntaREST/commit/2193e431d67d0f1807a2d1cc0853329871c63479))
*  **common:** add undo/redo to Form component ([fc02c00](https://github.com/AntaresSimulatorTeam/AntaREST/commit/fc02c00aea2c9b7d6cd9464346e8962d5304ca1c))
*  **ui:** enable undo/redo on some forms ([d0d07e9](https://github.com/AntaresSimulatorTeam/AntaREST/commit/d0d07e92b17151ef1c650b5cb6b12b67f3b6e6e3))
*  **ui-config:** enable undo/redo in adequacy patch form (#1543) ([19f2679](https://github.com/AntaresSimulatorTeam/AntaREST/commit/19f26794b0c7dc9d824c624bb90cc73b9e7cbf98))
*  **matrix:** improve matrix read/write using NumPy ([16b7403](https://github.com/AntaresSimulatorTeam/AntaREST/commit/16b7403265d0f61dcd2563076f865576ebb7db0a))


### Bug Fixes

*  **command:** fix update_config to support API PUT v1/studies/{uuid}/raw ([ffed124](https://github.com/AntaresSimulatorTeam/AntaREST/commit/ffed124cb17965e41409812a83da2bf36da8b014))
*  **command:** change code according to comments ([f3c68d2](https://github.com/AntaresSimulatorTeam/AntaREST/commit/f3c68d2d8cf323a36582a5e5a165c1ddf23c0d7c))
*  **api:** correct the regex in `IniReader` to match a section in square brackets ([123d716](https://github.com/AntaresSimulatorTeam/AntaREST/commit/123d716c68584c8c2ce5d907991c7379e21f0a4a))
*  **web:** change API response_model to avoid ValidationError (#1526) ([e42bac4](https://github.com/AntaresSimulatorTeam/AntaREST/commit/e42bac4eea13ea95efa52b2960d6810d79b6d810))
*  **ui-i18n:** add missing key ([685c487](https://github.com/AntaresSimulatorTeam/AntaREST/commit/685c487f0f73d8eb37b1594c5e46b283dd55d8d4))
*  **storage:** fix INI file writing anomaly in Antares study update (#1542) ([a786269](https://github.com/AntaresSimulatorTeam/AntaREST/commit/a786269d2597b6bf330bc85080361553faf1f8bd))
*  **variant-manager:** correct the variant manager ([5cc467f](https://github.com/AntaresSimulatorTeam/AntaREST/commit/5cc467f48a2f6b73d9abf652391c14bef76d40fc))


### Code Refactoring

*  **allocation:** remove deprecated class `FixedAllocationKeyIniReader` ([b1299fe](https://github.com/AntaresSimulatorTeam/AntaREST/commit/b1299fea4149a57a56bf4e1e002597d7f35a7625))
*  **rawstudy-filesystem:** improve error message raised by `ChildNotFoundError` ([9a6c4f9](https://github.com/AntaresSimulatorTeam/AntaREST/commit/9a6c4f92b704219836e42b15c196f4c317ee3fad))


### Tests

*  **ini:** correct the `IniFileNode` unit tests ([b96a92e](https://github.com/AntaresSimulatorTeam/AntaREST/commit/b96a92eadb6ca8b69b1bfc2e335a5535d67f52b1))
*  **api:** simplify initialization of the database in unit tests (#1540) ([e0a6000](https://github.com/AntaresSimulatorTeam/AntaREST/commit/e0a6000c7aa8fe8a8fb17642b6c2cf01bc95552f))
*  **matrix-service:** improve unit tests for `MatrixService` class ([8cf9f94](https://github.com/AntaresSimulatorTeam/AntaREST/commit/8cf9f94e8e5b79a966552e9728d50455b885b117))


### Styles

* reindent source code in unit tests ([4515451](https://github.com/AntaresSimulatorTeam/AntaREST/commit/4515451313b5ab3935dfbb360eedce794bfe5d3e))
* reindent `ini_reader.py` ([d5fa83d](https://github.com/AntaresSimulatorTeam/AntaREST/commit/d5fa83d7371b533a0b6aac38aac924c3e3d8a0de))


### Contributors

<a href="https://github.com/laurent-laporte-pro">laurent-laporte-pro</a>, <a href="https://github.com/hdinia">hdinia</a>, <a href="https://github.com/MartinBelthle">MartinBelthle</a>, <a href="https://github.com/skamril">skamril</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

laurent-laporte-pro committed:
build: remove unused libraries from project requirements

laurent-laporte-pro committed:
build: move `checksumdir` and `locust` libraries from project requirements to tests requirements (only used in unit tests)

laurent-laporte-pro committed:
build: include `typing_extensions` in the project requirements (new direct usage)

laurent-laporte-pro committed:
build: update project's metadata in `setup.py` (author, author email, license, platforms, classifiers)

laurent-laporte-pro committed:
build: update project configuration, specify exclusion of non-package directories when using `find_packages`

laurent-laporte-pro committed:
build: remove unused Sphinx-related libraries from docs requirements

laurent-laporte-pro committed:
fix(api): Correct endpoint for HTTPStatus.NO_CONTENT (204)

The response model has been updated to `None`.

laurent-laporte-pro committed:
test(api): correct JSON payloads for "table mode"

Corrects the JSON payloads for endpoints related to the "table mode" to properly convert enumerated values into strings.

laurent-laporte-pro committed:
fix(api): correct `update_area_ui` endpoint response model

The `update_area_ui` endpoint returns `None`.

laurent-laporte-pro committed:
test(api): correct unit tests for Xpansion-related endpoints to use bytes for file upload

Correct the unit tests for Xpansion-related endpoints to use `io.BytesIO` instead of `io.StringIO` for sending a file in multipart.

laurent-laporte-pro committed:
test(api): update integration tests for Starlette 0.21.0

Starlette 0.21.0 introduced changes to the `TestClient` implementation by replacing the `requests` library with `httpx`. This commit corrects and updates unit tests in the `integration` directory to accommodate the new API and resolve breaking changes.

laurent-laporte-pro committed:
refactor(tools): prepare for migration from `Requests` to `httpx`

This commit prepares for migrating from Requests to httpx by adjusting imports and usage in LDAP, command line tools, and GUI code.

laurent-laporte-pro committed:
test(api): update stream file download tests for compatibility with Requests and Httpx

laurent-laporte-pro committed:
fix(api): enhance Python detection in version info endpoint

laurent-laporte-pro committed:
test(api): update `UploadFile` object construction for `Starlette` >= 0.24

Adjusted the construction of `UploadFile` objects in unit tests (changes in Starlette's API): the 'content_type' parameter has been removed.

laurent-laporte-pro committed:
test(api): improve unit test of `get_dependencies` endpoint

laurent-laporte-pro committed:
ci: change the configuration of mypy to ignore `httpx` package

laurent-laporte-pro committed:
chore: correct indentation

sylvlecl committed:
fix(docs): update README, removing obsolete instructions (#1552)

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>
Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>

laurent-laporte-pro committed:
Merge pull request #1726 from AntaresSimulatorTeam/bugfix/1724-correct-api-endpoints-and-cli-tools

fix(api): correct API endpoints and CLI tools

MartinBelthle committed:
feat(service): assign to a copied study all groups of the user (#1721)

feat(service): assign to a copied study all groups of the user #1721

laurent-laporte-pro committed:
refactor(tools): correct migration from `Requests` to `httpx`

laurent-laporte-pro committed:
fix(api): sanitize and avoid duplicates in group IDs lists

laurent-laporte-pro committed:
docs(api): improve the documentation of the study upgrade and import endpoints

laurent-laporte-pro committed:
Merge pull request #1649 from AntaresSimulatorTeam/feat/set_public_mode_to_read_when_user_has_no_group

feat(service): set PublicMode.READ when importing is done by user with no groups

laurent-laporte-pro committed:
docs: update changelog for the v2.15 release

laurent-laporte-pro committed:
fix(matrix): create empty file if the matrix is empty

laurent-laporte-pro committed:
chore: correct handling of base class with no `__annotations__` in `AllOptionalMetaclass`

This correction addresses an issue where the `AllOptionalMetaclass` metaclass was not handling cases where the base class has no `__annotations__` attribute properly. In such situations, this fix ensures that an empty annotation dictionary is assumed for correct behavior.

laurent-laporte-pro committed:
fix(thematic-trimming): correct the name of the "Profit by plant" variable

This modification addresses the variable name used to enable or disable the calculation of power plant profits. It replaces the name "Profit" with "Profit by plant." On the back-office implementation side, changes have been made to simplify maintenance, and tests have been added to ensure proper handling of this variable. On the front end, the variable has been renamed to `profitByPlant` to align with TypeScript/ReactJS naming conventions.

laurent-laporte-pro committed:
test(thematic-trimming): correct unit test (`study.version` was not set)

laurent-laporte-pro committed:
Merge pull request #1742 from AntaresSimulatorTeam/bugfix/1741-correct-thematic-trimming-variable-profit-by-plant

laurent-laporte-pro committed:
refactor(matrix): improve implementation of dataframe saving

laurent-laporte-pro committed:
refactor(matrix-service): improve implementation of `create_by_importation`

laurent-laporte-pro committed:
feat(antares-launcher): use antares launcher v1.3.1 (#1743)

laurent-laporte-pro committed:
Merge pull request #1746 from AntaresSimulatorTeam/bugfix/fix-empty-array-dataframe-save

bugfix(matrix): correct the loading and saving of empty matrices as TSV files

a-zakir committed:
fix(ui-xpansion): add the missing `batch_size` field in the Xpansion parameters view (#1739)

laurent-laporte-pro committed:
build(deploy): update the GitHub deploy workflow to generate bin artefacts

laurent-laporte-pro committed:
docs: update changelog and release date

laurent-laporte-pro committed:
build(deploy): correct the GitHub deploy workflow to allow legacy dependencies

skamril committed:
fix(ui-debug): issue to display json file in outputs (#1747)

laurent-laporte-pro committed:
build: new minor release v2.15.0 (2023-09-30)

laurent-laporte-pro committed:
fix(packaging): update the packaging script to use Antares Solver v8.6.2

flomnes committed:
feat(desktop): enable year-by-year & synthesis output in study "000 Free Data Sample" (#1735)

Co-authored-by: Laurent LAPORTE <laurent.laporte.pro@gmail.com>

laurent-laporte-pro committed:
build: new minor release v2.15.0 (2023-09-30)

laurent-laporte-pro committed:
Revert "build(deploy): update the GitHub deploy workflow to generate bin artefacts"

This reverts commit acf3c07eb9959bada064fa2a29b5f5824980af1c.
